### ISSUE TITLE
[HOLD for payment 2023-10-27] [$500] Chat - Emoji are cut off at the top if message contains only emoji and is located under avatar

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#30064]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.87-8

**Reproducible in staging?:** Y

**Reproducible in production?:** N

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause-Internal Team

**Slack conversation:** @

## Action Performed:
1. Open iOS/iPad or Android App
2. Login with any account
3. Go to any chat
4. Send a message that contains only emoji

## Expected Result:
Emoji are displayed correctly in chat history

## Actual Result:
Emoji are cut off at the top if the message contains only emoji and is located under the avatar and sending time

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native/iPad 
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/115492554/f821cc16-ace1-43cc-9a88-0930697cfb83

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f9feff479bfdeec1</li>
        <li>Upwork Job ID: 1715303315170639872</li>
        <li>Last Price Increase: 2023-10-20</li>
    </ul>
</details>

**Comment by izarutskaya** [comment]
Not reproducible in production

https://github.com/Expensify/App/assets/115492554/44bcb646-ca37-4118-ab6c-88bfa8e07968

**Comment by jayeshmangwani** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Emoji are cut off at the top if the message contains only emoji(s)

### What is the root cause of that problem?
Issue is happening due to `Invisible Characters` that was added in this [PR](https://github.com/Expensify/App/pull/29686), in native devices android and iOS overwrite the `&#x200b;` Text styling, and in this case it is taking default parent Text styling and overwriting the `styles.onlyEmojisText` styling thats why it is taking default text fontSize and lineHeight instead of 30

https://github.com/Expensify/App/blob/69cf29e32950f915ea888fe294dfdea6a9fabbb5/src/pages/home/report/ReportActionItemFragment.js#L142

https://github.com/Expensify/App/blob/69cf29e32950f915ea888fe294dfdea6a9fabbb5/src/pages/home/report/ReportActionItemFragment.js#L103-L109

### What changes do you think we should make in order to solve the problem?
To the simplest solution I would suggest we should wrap the the `<Text>&#x200b;</Text>` in the Container `View` then issue will be solved in native devices and then styling is not being override 

like below
```javascript
<View><Text>&#x200b;</Text></View>
```

### What alternative solutions did you explore? (Optional)
1. we can add the emoji styling to the Invisible Characters text if it only include the emojis like below

```javascript
<Text  style={[containsOnlyEmojis ? styles.onlyEmojisText : undefined, styles.ltr, ...props.style, isPendingDelete ? styles.offlineFeedback.deleted : undefined]}>&#x200b;</Text>;
```

2. create a platform dependent Invisible Characters file that only pass to the web and desktop and for native it will pass null as we do not need the Invisible Characters in native devices as original issue is happening in web and desktop

**Comment by narefyev91** [comment]
@ayazalavi will you be able to fix this asap?

**Comment by jayeshmangwani** [comment]
If above [proposal](https://github.com/Expensify/App/issues/30064#issuecomment-1772540962) is valid, then I can work on this, as the issue is already external, and I am available for PR

@narefyev91 @s77rt

**Comment by ayazalavi** [comment]
> @ayazalavi will you be able to fix this asap?

yes sure

**Comment by narefyev91** [comment]
@jayeshmangwani based on the process - if that was identified as regression from other PR - author of that PR should make changes to resolve the issue

**Comment by narefyev91** [comment]
Pinging @Li357 as well here

**Comment by jayeshmangwani** [comment]
> based on the process - if that was identified as regression from other PR - author of that PR should make changes to resolve the issue

@narefyev91 Yes, I am aware of that process, but I was asking as this issue has a `Help Wanted` tag, and I proposed a solution when the issue was open for proposal, but yes, I will keep it in mind for next time to not propose a solution when the issue is regression from some other PR

**Comment by narefyev91** [comment]
> > based on the process - if that was identified as regression from other PR - author of that PR should make changes to resolve the issue
> 
> @narefyev91 Yes, I am aware of that process, but I was asking as this issue has a `Help Wanted` tag, and I proposed a solution when the issue was open for proposal, but yes, I will keep it in mind for next time to not propose a solution when the issue is regression from some other PR

I really appreciated your help here - and for sure your solution is correct - i will address that your import was very important here - and will ask to pay you a bonus after the issue will be resolved

**Comment by narefyev91** [comment]
@ayazalavi how much time you need to create a PR fix?

**Comment by ayazalavi** [comment]
PR: https://github.com/Expensify/App/pull/30081

**Comment by s77rt** [comment]
Unassigning since I'm not the C+ here. @sakluger Please assign @narefyev91 and @ayazalavi

**Comment by sakluger** [comment]
Hey @narefyev91 and @jayeshmangwani, I just discussed this case with the team. It's true that in general, when there is a regression, we want the author of the offending PR to fix it. But when it comes to deploy blockers, because of the urgency, we will accept proposals from other contributors if it allows us to get a fix out quicker.

In this case, because @jayeshmangwani proposed the fix we ended up using, we can pay them the full $500 .

**Comment by sakluger** [comment]
Thanks for the quick work everyone! I don't think any additional payment is due here since this was a regression from https://github.com/Expensify/App/pull/29686, and the same contributors worked on the fix as the original issue, so payment will be handled in the original issue.

**Comment by jayeshmangwani** [comment]
> 
> In this case, because @jayeshmangwani proposed the fix we ended up using, we can pay them the full $500 .

@sakluger payment is due for me , can you please check here ?

**Comment by sakluger** [comment]
@jayeshmangwani Sorry for closing before paying out and thanks for letting me know. I've send you an offer via Upwork, please let me know once you've accepted.

**Comment by sakluger** [comment]
Awesome, thanks you - I've completed payment. Sorry again for closing prematurely!
